src/senaite/core/workflow/__init__.py

```diff
@@ -18,10 +18,214 @@
 # Copyright 2018-2024 by it's authors.
 # Some rights reserved, see README and LICENSE.
 
+from bika.lims import api
+from Products.DCWorkflow.DCWorkflow import DCWorkflowDefinition
+from Products.DCWorkflow.Guard import Guard
+from senaite.core import logger
+
 ANALYSIS_WORKFLOW = "senaite_analysis_workflow"
 DUPLICATE_ANALYSIS_WORKFLOW = "senaite_duplicateanalysis_workflow"
 REFERENCE_ANALYSIS_WORKFLOW = "senaite_referenceanalysis_workflow"
 REFERENCE_SAMPLE_WORKFLOW = "senaite_referencesample_workflow"
 REJECT_ANALYSIS_WORKFLOW = "senaite_rejectanalysis_workflow"
 SAMPLE_WORKFLOW = "senaite_sample_workflow"
 WORKSHEET_WORKFLOW = "senaite_worksheet_workflow"
+
+_marker = object()
+
+
+def get_workflow(workflow_or_id, default=_marker):
+    """Returns the DCWorkflow object for the given id or workflow
+    """
+    if isinstance(workflow_or_id, DCWorkflowDefinition):
+        return workflow_or_id
+
+    # Look-up the workflow by id
+    wf_tool = api.get_tool("portal_workflow")
+    workflow = wf_tool.getWorkflowById(workflow_or_id)
+    if not workflow:
+        if default is _marker:
+            raise ValueError("Workflow not found: %s " % workflow_or_id)
+        return default
+
+    return workflow
```

Comment:
Would be cool to have this function accept the object as well to return the primary workflow;)